### ISSUE TITLE
`self-update` does not install over existing version

### ISSUE THREAD

**Issue by feinstruktur** [qutheory/vapor-cli#12]
I noticed that only a part of my [change for the `self-update` location](https://github.com/qutheory/vapor/pull/258) came across to this repo. What was lost was the in-place update.

In other words, right now when you are in dir `foo` and have `vapor` in `~/bin/` you'll end up with `foo/vapor` and the one in `~/bin` is unchanged.

The fix is simple, use `Process.arguments[0]` instead of `directory` when moving the downloaded file.

Since I'm proposing to do a compile instead of a move in #10 I could do both in one go.

**Comment by tannernelson** [comment]
Directory should be equal to `Process.arguments[0]`

```swift
var iterator = Process.arguments.makeIterator()

guard let directory = iterator.next() else {
    fail("no directory")
}
```

That is the code that makes it.

**Comment by feinstruktur** [comment]
Ah right, of course. The issue instead is when you’ve got `vapor` in your path. You don’t get the full path then:
```
$ which vapor
/Users/sas/bin/vapor
$ ~/bin/vapor self-update
Process.arguments: ["/Users/sas/bin/vapor", "self-update"]
$ vapor self-update
Process.arguments: ["vapor", "self-update”]
```

**Comment by feinstruktur** [comment]
This is a weird issue. So what’s happening is that depending on wether `vapor` is a script or a binary, the process arguments are different. In the first case, it’s a full path, in the latter case it’s just `vapor`.

What I’m going to do is test for that and substitute `which vapor` in the latter case:

```
            let target: String = {
                if orig == "vapor" {
                    do {
                        return try runWithOutput("which vapor")
                    } catch {
                        fail("Command was called with 'vapor' but is not in path")
                    }
                } else {
                    return orig
                }
            }()
```

I’m pretty sure this is related to the self-update issue we spoke about in slack.

**Comment by tannernelson** [comment]
`which vapor` seems like a good solution. Maybe we should just run that for all cases to keep the code simple?

**Comment by feinstruktur** [comment]
I don't think that's a good idea, because I've got vapor in `~/bin/vapor` with high precedence for example but I might want to compile or update `/usr/local/bin/vapor`. I've got a fix/change for this, will push in a sec.

**Comment by feinstruktur** [comment]
Fixed by b35058c49f8b4a9ac2ebe30e3c120a8d01e91996
